### ISSUE TITLE
Review and adjust the search form page for accessibility

### ISSUE THREAD

**Issue by Captainkirkdawson** [FreeUKGen/MyopicVicar#946]
As the main landing page and the prime input page of the application this page must conform with accessibility recommendations.
Dependant on #945

**Comment by Captainkirkdawson** [comment]
Alison has mode a first effort at addressing the search page.
 It is available for first viewing at 
https://test3.freereg.org.uk/cms/sandbox-for-alison

Remember this is not an operational page

**Comment by Captainkirkdawson** [comment]
I like the changes being proposed. 
1)Unable to get keyboard only county selection working on any browser
2)Am unsure of what focus is (my ignorance) in the "Search button can be tabbed to but there is currently no visual indication of focus (because "ladda" styling only specifies hover state) — other buttons, including submit, get hover state styling without any intervention by me" The search button for this pages is actually a javascript link to provide the processing indication. Am unsure what is actually required. 
3) Are tooltips accessible to screen reader users?. I suggest we ask someone
4)The presentation of Record type is potentially confusing (see Shaw report). Could you point me to where the issue is commented upon?
5)Are unique alt attributes on "i" icons still necessary?. Unsure of issue
6) Should we consider providing real-time, always-visible feedback on which/how many counties are selected? Counties that are adjacent geographically could be a long way apart in the select listing. Valid observation but how would we do that?

**Comment by AlOneill** [comment]
1) **Select boxes** for keyboard-only users are complicated! Especially multiple-select. The alternative is to use a set of check-boxes, but this would only be feasible for Record type.

Keyboard-only selection for County works by first tabbing into select box, then hold ctrl (cmd on Mac) whilst using arrow keys to navigate and *spacebar* to select/unselect. if on IE8+, then hit shift+F8 first to enter correct 'mode'.

Mac users can use ctrl+F7 to toggle tabbing behaviour between "All" and form controls only. (If on a Mac OS X that uses Spotlight, then cmd+spacebar shortcut for Spotlight will need changing: I changed to shift+cmd+spacebar).

2) **Focus** can be thought of as the keyboard-only equivalent of hover, triggered by tabbing to link or another tab-able  element.

3) and 5) Graham Rees-Evans (Shaw Trust) is away until 26 Sep -- my email is awaiting his return. 5 was originally associated with use of links for tooltips, but we are removing links from form labels (STAS-F11)

4) See page 26, 2nd para from bottom -- this was a user who could not access tooltip text for a fuller explanation.

6) No real idea, just reflecting on my own experience, but JavaScript would be involved ...

**Comment by Captainkirkdawson** [comment]
WRT 1) Select boxes for keyboard-only users are complicated! Especially multiple-select. The alternative is to use a set of check-boxes, but this would only be feasible for Record type.
 Why do you feel this is not feasible? A mobile device already displays the selection of counties and places as check boxes so I think we could consider that universally if it makes accessibility better

**Comment by AlOneill** [comment]
Sorry, I should have said radio buttons -- Record type is a choose one.

Chrome emulator does not show what you describe, Kirk. Still see select boxes. Are you viewing pages on an emulator or a real device? If an emulator, which one? If a real device, which one?

**Comment by AlOneill** [comment]
Have experimentally replaced **Record type** select with a radio group on [https://test3.freereg.org.uk/cms/sandbox-for-alison]. Please note that on a *real* form, the page would load initially with "All three types" selected (Refinery editor doesn't allow this). Keyboard-only use of radio group is much simpler than for a select element: tab into group, then up/down arrow keys to navigate/choose.

Re mobile emulators: it seems that the free ones are unable to replace select elements with check boxes.

**Comment by Captainkirkdawson** [comment]
Have committed  to test3 Alison's latest version of the search page with accessibility changes implemented
